### PR TITLE
ML Model Benchmarking — TB Treatment Outcome Prediction (CAR, 2015–2025)

### PR DESCRIPTION
**Pull request by Benny-Gil**
## What does this PR do?
Baseline Models Notebook contains benchmarking 12 ML models for binary classification of TB treatment outcomes using ML ready dataset. This is to identify top performing models for our non-temporal dataset.

Improved Models Notebook contains top 5 models where SMOTE was applied to address class imbalance, this also includes additional feature (Diagnosis to Treatment Delay).

## Related Issue
Closes #18 

## Type of Change
- [x] Feature
- [ ] Bug fix
- [ ] Improvement
- [ ] Technical / Infrastructure

## How was this tested?
- [x] Local dev
- [x] Manual testing
- [ ] Automated tests (if applicable)

### Summary
**Random Forest +SMOTEENN** — best overall AUC, F1, and generalisation.  

| | |
|---|---|
| **Class split** | 86.2% Success / 13.8% Failure |
| **Data split** | 70% Train / 20% Test / 10% Validation (stratified) |
| **Features** | 15 total — 4 new geographic/facility + 1 derived delay feature |
| **Resampling** | SMOTE+ENN on training only → Failure class +433%, near-balanced 0.70:1 ratio |

---

### Test Set (20%)

| Model | AUC | Recall (Fail) | F1 (Fail) |
|-------|-----|--------------|----------|
| **Random Forest**  | **0.7154** | **57.98%** | **0.3539** |
| Gradient Boosting | 0.7121 | 44.68% | 0.3262 |
| LightGBM | 0.7119 | 48.40% | 0.3467 |
| XGBoost | 0.7095 | 47.87% | 0.3416 |
| Logistic Regression | 0.6816 | 77.13%* | 0.3030 |

*\*High recall but only 19.6% precision and ~51% accuracy — over-alerts frequently*

### Validation Set (10% held-out)

Random Forest confirmed best: **Val AUC 0.7380**, Val Recall (Fail) **52.13%**  
All models showed positive Test→Val AUC gains — no overfitting to test set.

---

## SMOTE+ENN vs Baseline

| Model | ΔAUC | ΔRecall (Fail) |
|-------|------|---------------|
| Random Forest | **+0.045** | +27.1% (31% → 58%) |
| Gradient Boosting | +0.024 | +39.9% (5% → 45%) |
| LightGBM | +0.033 | −5.3% |
| XGBoost | +0.028 | −4.2% |
| Logistic Regression | −0.010 | +11.7% |

---

## Top Features

1. `Age` · `Days_To_Treatment` — strongest predictors
2. `Year` — reflects protocol changes over 2015–2025
3. `City/Municipality` · `Treatment Health Facility` — facility/geographic variation
4. `Diagnosis_to_Treatment_days` *(new)* — clinically meaningful delay indicator

### CODE REVIEW

**Comment by Benny-Gil** [comment]
Ranking | Model | Why
-- | -- | --
1 (for now) | Logistic Regression | Highest true discrimination, interpretable coefficients, balanced recall, fast
2 | SVM (RBF) | Marginally higher AUC but slower, black-box
Avoid | AdaBoost, MLP, GBM | High F1 only because they predict nearly everything as Success — useless for catching failures

**Review by VladTemp27 — CHANGES_REQUESTED** [review]
Retrain this using the updated dataset, include Training time of each model and how many epochs

**Comment by Benny-Gil** [comment]
@VladTemp27 @georcelle `./non-temporal/improved_models_with_smote_enn.ipynb` contains the best result based on my experimentation with different sampling techniques. You can disregard all other notebook files except for `baseline_models.py` which contains initial model benchmarking

**Review by georcelle — APPROVED** [review]
checked the result visuallizations and was used for the poster

**Review by Aizakkuuu — APPROVED** [review]
lgtm
